File: javascript/108-convert-sorted-array-to-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
 var sortedArrayToBST = function(nums) {
    if (!nums.length) return null
    const mid = Math.floor((nums.length-1) / 2)
    const node = new TreeNode(
        nums[mid],
        sortedArrayToBST(nums.slice(0, mid)),
        sortedArrayToBST(nums.slice(mid + 1))
    )
    return node
};

File: golang/108-convert-sorted-array-to-binary-search-tree.go
package golang

import "math"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func sortedArrayToBST(nums []int) *TreeNode {
    if len(nums) == 0 {
        return nil
    }
    mid := int(math.Floor(float64((len(nums)-1) / 2)))
    return &TreeNode{
        nums[mid],
        sortedArrayToBST(nums[0:mid]),
        sortedArrayToBST(nums[mid+1:]),
    }
}